docs: add overlays docs (#4)
siderolabs/overlays: README.md

| jetson_nano            | Jetson Nano                     | [sbc-jetson](https://github.com/siderolabs/sbc-jetson)           |
| bananapi_m64           | BananaPi M64                    | [sbc-allwinner](https://github.com/siderolabs/sbc-allwinner)     |
| libretech_all_h3_cc_h5 | LibreTech H3 CC H5              | [sbc-allwinner](https://github.com/siderolabs/sbc-allwinner)     |
| pine64                 | Pine64 A64                      | [sbc-allwinner](https://github.com/siderolabs/sbc-allwinner)     |

## Building Overlays

A new overlay can be created using the [SBC Template](https://github.com/siderolabs/sbc-template) repository as a starting point and
replacing all references of `board` with  your SBC name.

Then followed by running `make rekres` to automatically generate/update all the required files.
Then the user can run `make help` on instructions on how to setting up `docker buildx`.

> Note: Run `make rekres` after any changes to the `.kres.yaml` file or before committing any changes.

It is highly recommended to take a look at an existing overlays from the [catalog](#overlays-catalog) as a template for building your own.

> ### Folder structure

Reviewer: nit:

```suggestion
### Folder Structure
```

│       │   └── rpi_4
│       ├── firmware
│       │   └── rpi4
│       │       └── firmware_20240302.bin
│       └── u-boot
│           └── rpi4.uboot.img
├── installers
│   └── rpi_generic
└── profiles
    ├── rpi_4.yaml
    └── rpi_5.yaml

12 directories, 3 files
```

An overlay can provide multiple `installers` and `profiles` for a group of SBC's.

The `artifacts` folder contains the files that will be copied to the Talos boot and installer assets.
This is an optional folder, if the overlay does not require any artifacts to be copied to the boot asset, it can be omitted.

The `installers` folder contains statically linked binaries named after the profile names.
If an overlay provides only a single profile it can be named `default`.

The `profiles` folder contains the list of profiles that the overlay supports.
Talos Imager will register profiles based on the file names in this folder.

### Installer implementation

The installer is a statically linked binary that is responsible for providing the logic Talos Imager will consume to generate a boot asset.

Installer can be any statically linked binary and has to be provided for both `amd64` and `arm64` architectures.

> SideroLabs will provide an official Go wrapper to handle the communication between the installer and Talos Imager.

Reviewer: ```suggestion
Sidero Labs will provide an official Go wrapper to handle the communication between the installer and Talos Imager.
```

> Siderolabs provides an [`adapter`](https://pkg.go.dev/github.com/siderolabs/talos/pkg/machinery/overlay/adapter#Execute) package than can used as the entrypoint of the Go program.

Reviewer: ```suggestion
Sidero Labs provides an [`adapter`](https://pkg.go.dev/github.com/siderolabs/talos/pkg/machinery/overlay/adapter#Execute) package than can used as the entrypoint of the Go program.
```

The Go code should implement the [`overlay.Installer`](https://pkg.go.dev/github.com/siderolabs/talos/pkg/machinery/overlay#Installer) interface.

An example of a simple installer can be found in the [sbc-template](https://github.com/siderolabs/sbc-template/blob/main/installers/board/src/main.go).

Refer to using [Custom installers](#custom-installers) for more information on how to use custom installers.

> #### Custom installers

Reviewer: ```suggestion
#### Custom Installers
```
